### PR TITLE
Added Geometrical parameters to nuclear.py

### LINKED ISSUE #489: New method: calculate all possible distances, angles, dihedrals
As discussed on the dev mailing list, we should have a method that calculates all possible distances, angles, dihedrals, etc.

Possibly useful references:
* https://github.com/CrawfordGroup/ProgrammingProjects/tree/master/Project%2301
* https://github.com/berquist/programming_party/blob/master/eric/molecule.py
* https://stackoverflow.com/q/20305272/3249688

From Karol:
> Regarding the coordinate-to-angle calculations - they'll be so fast for 99% of the cases that it's not worthwhile to think about calculating things on-the-fly.

So things should be calculated all at once.

### PR DESCRIPTION
**Pull request by GomaruSai**
Added distance, angle and dihedral methods to nuclear.py.
This will parially resolve #489
What's left is to figure out a way to represent this data in the final output file.

### CODE REVIEW

File: cclib/method/nuclear.py [hunk 1]
```diff
@@ -100,6 +100,91 @@ def stoichiometry(self):
             sign = "+" if self.data.charge > 0 else "-"
             formula += "(%s%i)" % (sign, magnitude)
         return formula
+    
+    def get_distances(self, atomcoords_index = -1):
```
Comment (langner) [inline cclib/method/nuclear.py:104]: We should just call this `distances` to keep in line with `stiochiometry` in this class and the no-get pattern we have in other places.

File: cclib/method/nuclear.py [hunk 2]
```diff
@@ -100,6 +100,91 @@ def stoichiometry(self):
...
+        '''Returns a matrix with distances between i_th and j_th atoms at index [i][j], where i < j
+        *Other indices will give wrong values*'''
```
Comment (langner) [inline cclib/method/nuclear.py:106]: What do you mean by wrong?

File: cclib/method/nuclear.py [hunk 3]
```diff
@@ -100,6 +100,91 @@ def stoichiometry(self):
...
+        coords = self.data.atomcoords[atomcoords_index]
+        number_of_atoms = self.data.atomcoords.size[1]
```
Comment (langner) [inline cclib/method/nuclear.py:108]: Why not use `natom`?

File: cclib/method/nuclear.py [hunk 4]
```diff
@@ -100,6 +100,91 @@ def stoichiometry(self):
...
+        distance_matrix = np.empty((number_of_atoms, number_of_atoms))
+        i = 0
+        j = 1
+        while(i < number_of_atoms):
```
Comment (langner) [inline cclib/method/nuclear.py:112]: In Python it is more more conventional to use `for` loops for these types of things.

File: cclib/method/nuclear.py (hunk 1 above)
Comment (GomaruSai) [reply cclib/method/nuclear.py]: Yes. I guess that makes sense. I have made changes in the new commit.

File: cclib/method/nuclear.py (hunk 2 above)
Comment (GomaruSai) [reply cclib/method/nuclear.py]: I used np.empty and did not populate the matrix at those indices. However, I have resolved this in the new commit. There is no longer any restriction in the indices to be used, unless it means the same thing.

File: cclib/method/nuclear.py (hunk 3 above)
Comment (GomaruSai) [reply cclib/method/nuclear.py]: Yes. I guess that makes sense. I have made changes in the new commit.

File: cclib/method/nuclear.py (hunk 4 above)
Comment (GomaruSai) [reply cclib/method/nuclear.py]: Yes. I guess that makes sense. I have made changes in the new commit.

File: cclib/method/nuclear.py (hunk 2 above)
Comment (GomaruSai) [reply cclib/method/nuclear.py]: *as long as it means the same thing

**Comment by berquist** [comment]
Since there was never any response on this, in the interest of letting someone else work on #489, I'll close it.

**Pull request closed without merging**